azure-pipelines.yml

```diff
@@ -184,14 +184,68 @@ jobs:
 
 # Build Correctness Jobs
 
-- job: Correctness_Determinism
+- template: eng/pipelines/evaluate-changed-files.yml
+  parameters:
+    jobName: Determine_Changes
+    vmImageName: ubuntu-latest
+    paths:
+      - subset: compilers
+        include:
+        - src/Compilers/*
+        - src/Dependencies/*
+
+- job: Correctness_Build_Artifacts
+  dependsOn: Determine_Changes
   pool:
     name: NetCore1ESPool-Public
     demands: ImageOverride -equals Build.Windows.Amd64.VS2022.Pre.Open
   timeoutInMinutes: 90
+  variables:
+    compilerChange: $[dependencies.Determine_Changes.outputs['SetPathVars_compilers.containsChange']]
   steps:
     - template: eng/pipelines/checkout-windows-task.yml
 
+    - task: PowerShell@2
+      displayName: Restore
+      inputs:
+        filePath: eng/build.ps1
+        arguments: -configuration Release -prepareMachine -ci -restore -binaryLog
+
+    - task: PowerShell@2
+      displayName: Build
+      inputs:
+        filePath: eng/build.ps1
+        arguments: -configuration Release -prepareMachine -ci -build -pack -publish -sign -binaryLog
+
+    - script: $(Build.SourcesDirectory)\artifacts\bin\BuildBoss\Release\net472\BuildBoss.exe  -r "$(Build.SourcesDirectory)/" -c Release -p Roslyn.sln
```

Comment:
Shouldn't we run BuildBoss on most PRs? It verifies more than compiler packages.

Reply:
I should always run. I believe it doesn't have any conditions on it